dias/outfile_tracking.py

```diff
@@ -0,0 +1,183 @@
+            else:
+                raise DiasException(
+                    "Type of additional value {}: {} is not supported (yet).".format(
+                        name, type(default)
+                    )
+                )
+            self._create_cmd += "{} {}, ".format(name, typename)
+            self._insert_cmd += "?, "
+        self._create_cmd += "filename TEXT UNIQUE ON CONFLICT REPLACE)"
+        self._insert_cmd += "?)"
+
+        # Open connection to data index database
+        # and create table if it does not exist.
+        db_file = os.path.join(self.state_dir, self._DB_FILE)
+        db_types = sqlite3.PARSE_DECLTYPES | sqlite3.PARSE_COLNAMES
+
+        # The connection can only be used by one thread at a time.
+        # This module does not serialize use of the connection.
+        # Don't schedule multiple tasks at the same time.
+        # At the moment dias doesn't support that anyways.
+        self.data_index = sqlite3.connect(
+            db_file, detect_types=db_types, check_same_thread=False
+        )
+
+        cursor = self.data_index.cursor()
+        cursor.execute(self._create_cmd)
+        self.data_index.commit()
+
+    def __del__(self):
+        self.data_index.close()
+
+    def get_start_time(self, refresh_db=True):
+        """
+        Get the next start time.
+
+        Parameters
+        ----------
+        refresh_db : bool
+            Refresh the database before getting the start time (default: True).
+
+        Returns
+        -------
+        UTC date and time of next start time.
+        """
+        if refresh_db:
+            self._refresh_data_index()
+        cursor = self.data_index.cursor()
+        query = "SELECT stop FROM files ORDER BY stop DESC LIMIT 1"
+        results = list(cursor.execute(query))
+        if results:
+            return results[0][0]
+        return None
+
+    def update_data_index(self, start, stop, filename=None, **kwargs):
+        """Add row to data index database.
+
+        Update the data index database with a row that
+        contains the name of the file and the span of time
+        the file contains.
+
+        Update the data index database with a row that
+        contains a span of time, the relative path to the
+        output file and any additional values defined in the constructor call.
+
+        Parameters
+        ----------
+        start : unix time
+            Earliest time contained in the file.
+        stop : unix time
+            Latest time contained in the file.
+        filename : str
+            Name of the file.
+        **kwargs
+            Additional values to save. The keys have to be identical to what has been
+            defined in the constructor.
+
+        """
+        # check additional variables
+        for k in kwargs:
+            if k not in self._additional_vars.keys():
+                raise DiasException(
+                    "Found unexpected key in additional variables: {}".format(k)
```

Comment:
Do we also want to assert that all of the keys that are in `self._additional_vars.keys()` are included? Is this not an equality test?